modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -420,4 +607,97 @@ private void maybeScheduleJob() {
     DataLifecycleErrorStore getErrorStore() {
         return errorStore;
     }
+
+    /**
+     * This is a ClusterStateTaskListener that writes the force_merge_completed_timestamp into the cluster state. It is meant to run in
+     * STATE_UPDATE_TASK_EXECUTOR.
+     */
+    static class UpdateForceMergeCompleteTask implements ClusterStateTaskListener {
+        private final ActionListener<Void> listener;
+        private final String targetIndex;
+        private final ThreadPool threadPool;
+
+        UpdateForceMergeCompleteTask(ActionListener<Void> listener, String targetIndex, ThreadPool threadPool) {
+            this.listener = listener;
+            this.targetIndex = targetIndex;
+            this.threadPool = threadPool;
+        }
+
+        ClusterState execute(ClusterState currentState) throws Exception {
+            logger.trace("Updating cluster state with force merge complete marker for {}", targetIndex);
```

Comment:
minor change, since TRACE logs are difficult to retrieve:
```suggestion
            logger.debug("Updating cluster state with force merge complete marker for {}", targetIndex);
```